mixnet/node/src/lib.rs

```diff
@@ -77,38 +81,66 @@ impl MixnetNode {
     }
 
     async fn handle_connection(
-        mut socket: TcpStream,
+        socket: TcpStream,
+        cache: ConnectionPool,
         private_key: PrivateKey,
         client_tx: mpsc::Sender<Body>,
-    ) -> Result<(), Box<dyn Error>> {
-        let body = Body::read(&mut socket).await?;
+    ) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+        tokio::pin!(socket);
+        loop {
+            tokio::select! {
+                // TODO: add a random delay?
+                _ = tokio::time::sleep(std::time::Duration::from_millis(100)) => {}
+                body = Body::read(&mut socket) => {
+                    match body {
+                        Ok(body) => if let Err(e) = Self::handle_body(body, &cache, &private_key, &client_tx).await {
+                            tracing::error!("failed to handle body: {e}");
+                        },
+                        Err(e) => {
+                            tracing::error!("failed to read body: {e}");
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
We may need to give up reading this socket at the line 100, instead of continuing the loop?

Also, it seems we don't need the sleep. Please let me know if I'm missing something.

Reply:
Yeah, after rethinking, we do not need the sleep.